packages/react-forms/src/InputDate/index.js

```diff
@@ -21,141 +21,173 @@ function toYYYYMMDD(date) {
   return `${year}-${month}-${day}`;
 }
 
-type Props = {
+export type Props = {
   value: string,
   onChange: Function,
   isOpen?: boolean,
   onToggle?: Function,
   min?: string,
   max?: string,
   disabled?: boolean,
+  onCalendarChange?: Date => void,
+  calendarValue?: Date,
 };
-
-type State = {
-  current?: Date,
-  isOpen: boolean,
-};
-
-export default class InputDate extends React.Component<Props, State> {
-  component = React.createRef /*:: <HTMLElement> */();
-  refA = React.createRef /*:: <HTMLElement> */();
-  refB = React.createRef /*:: <HTMLElement> */();
-
-  state = {
-    current: undefined,
-    isOpen: false,
-  };
-
-  get isOpen(): boolean {
-    return Boolean(this.props.onToggle ? this.props.isOpen : this.state.isOpen);
-  }
-
-  setIsOpen = (isOpen: boolean) =>
-    this.props.onToggle
-      ? this.props.onToggle(isOpen)
-      : this.setState({ isOpen });
-
-  handleOpen = () => this.isOpen === false && this.setIsOpen(true);
-
-  handleClose = /* istanbul ignore next */ () =>
-    void (this.isOpen === true && this.setIsOpen(false));
-
-  handleCloseOnBlur = (evt: MouseEvent) => {
-    const { relatedTarget } = evt;
-    // istanbul ignore next (this is too DOM related)
-    if (
-      this.component.current && relatedTarget instanceof HTMLElement
-        ? !this.component.current.contains(relatedTarget)
-        : true
-    ) {
-      this.setIsOpen(false);
+function isBoolean(bool) {
+  return bool === true || bool === false;
+}
+const InputDate = ({
+  value,
+  onChange,
+  isOpen: isOpenProp,
+  onToggle,
+  min,
+  max,
+  onCalendarChange,
+  calendarValue,
+  ...props
+}: Props) => {
+  const component = createRef();
+  const refA = createRef();
+  const refB = createRef();
+
+  const [current, setCurrent] = useState(calendarValue);
```

Comment:
Here we should use https://www.npmjs.com/package/@quid/react-use-controlled-state, same for the `isOpen` property, and any property that the user can control via prop, but that would make sense to also handle internally.

Basically, we want to make this component usable in two ways:

1. You pass to it just `defaultX` values, and then the component updates its state internally, calling `onChange` callbacks to notify the user (optionally)
2. You pass it a value, and a mandatory `onChange` callback property. The component doesn't hold any state internally, but relies exclusively on the values provided by properties to the component.

If anything isn't clear feel free to contact me privately and we can look at it together.

Reply:
Okey, I added it where I found reason to use it.